pkg/pillar/cmd/README.md

```diff
@@ -0,0 +1,273 @@
+# Agent Services
+
+This directory contains agent services for pillar.
+
+Each subdirectory contains exactly one agent, whose name is the name of the directory.
+
+Look at [verifier](./verifier/) for an example of the implementation.
+
+## Design Pattern
+
+The design pattern for each agent is as follows. When creating a new agent, ensure
+that you follow the design pattern.
+
+The package name in the agent's directory is the same as the directory name.
+It should contain at least the following files:
+
+* `README.md` describing the agent and its purpose.
+* `run.go` containing the main library entry point for the agent.
+
+In addition, it should contain the following subdirectories:
+
+* [`lib/`](./lib/) contains core logic, has no dependency on being called command-line or pubsub, unless pubsub is part of the core business logic. It is important to keep these functions as simple and straightforward as possible. Unit tests of business logic should be here as well.
+* [`pubsub/`](./pubsub/) contains functions to handle pubsub messages, including all handlers. There should be no business logic here, only pubsub message handling. For business logic, handlers should call business logic functions in `lib/`.
+* [`cmd/`](./cmd/) contains CLI commands and sub-commands, all in package `main`. Can be called as `go run ./cmd`. Call either functions in `pubsub/` or `lib/` depending on the command.
+* [`run.go`](./run.go) is the entrypoint for calling as part of zedbox, exposes the implementation of [`types.AgentRunner`](https://pkg.go.dev/github.com/lf-edge/eve/pkg/pillar/types#AgentRunner).
```

Comment:
Should we add note here that it's desirable architecture, however if it doesn't fit changes are possible? I'm referring here to metadata server (in current iteration of refactoring :art: )

Reply:
Yeah. I'm away from laptop, but I'll add that later.

Doesn't metadata server follow that pattern as well? It has a `Run()`, etc. It _could_ be called from zedbox, as it meets the interface requirements. It isn't, as we call it from inside zedrouter. But it still fits the design pattern?

Reply:
It does, except for folder structure

Reply:
Given how long it takes for CI to run, I will leave this for now. I am happy to open a tiny PR just for adding this to docs afterwards.